### PR TITLE
Add headache support

### PR DESCRIPTION
**Pull request by pfernique**

### CODE REVIEW

File: misc/src/openalea/misc/headache.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  configparser.read(rootpath/'metainfo.ini')
+  config = dict(configparser.items('metainfo'))
+
+  headerhandler = NamedTemporaryFile(delete=False)
+  try:
+    headerhandler.file.write(tab+config['project']+'.'+config['package'])
+    if not config['description'] == '':
+      headerhandler.file.write(': '+config['description'])
+    headerhandler.file.write('\n'+tab+'\n')
+    headerhandler.file.write(tab+'Copyright 1995- '+str(datetime.now().year)+' Inria, Cirad, Inra')
```
Comment (gbaty) [inline misc/src/openalea/misc/headache.py]: We should use current year, 1995 is a bit outdated

File: misc/src/openalea/misc/headache.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  confighandler = NamedTemporaryFile(delete=False)
+  confighandler.file.write('| \".*\\\\.h\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.hpp\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.c\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.cpp\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.py\" -> frame open:\"#\" line:\"#\" close:\"#\"')
+  confighandler.close()
+
+  os.system('headache -c '+confighandler.name+' -h '+headerhandler.name+' '+' '.join([file for file in (rootpath/'src').walkfiles() if any([file.fnmatch(pattern) for pattern in patterns])]))
```
Comment (gbaty) [inline misc/src/openalea/misc/headache.py]: Python [os.system doc](https://docs.python.org/2/library/os.html#os.system) suggests to use subprocess.
See [Replacing Older Functions with the subprocess Module](https://docs.python.org/2/library/subprocess.html#replacing-os-system)

**Comment by gbaty** [comment]
Nice !
We want to follow "PEP8" for coding style, and we would like to check it automatically using travis. Idea is that all new file are pep8 compliant and older wil be migrated over time.
flake8 or pep8 utilities are very convenient to check that.
We follow all pep8 conventions excepted line length that we set to 120 instead of 80.

File: misc/src/openalea/misc/headache.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
+"""Script to automatically generate headers in files of the openalea and vplants modules.
+"""
+
+import os
+from types import ModuleType
+from collections import Sequence
+from tempfile import NamedTemporaryFile
+from ConfigParser import ConfigParser
```
Comment (gbaty) [inline misc/src/openalea/misc/headache.py]: ConfigParser is also imported on line 52

Comment (pfernique) [inline misc/src/openalea/misc/headache.py]: Now deleted on line 49

**Pull request merged**